### DISCUSSION TITLE
PAT or SSH knows me but permission denied

### DISCUSSION THREAD

**Discussion by RenateUSB** [community/community#76722] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

I've tried to push a branch using PAT or SSH. I can make either recognize my user name (without explicitly saying it).
```
git push origin mybranch
ERROR: Permission to Hagb/decryptBooxUpdateUpx.git denied to RenateUSB.
git remote -v
origin  https://ghp_bunchofstuff@github.com/Hagb/decryptBooxUpdateUpx.git
```
I'm using the same remote strings that I used to clone with (in two separate dirs, PAT and SSH)
The PAT is enabled with everything clicked.
I've tried everything a bunch of times.
Before that I got the warning that I couldn't use a password.
I can `git remote set-url` and (intentionally) misspell the PAT and it fails and asks for the password.
When I supply the correct password manually it gives the same error.

**Comment by lotirium** [comment]
The error message indicates that the username _RenateUSB_ does not have the necessary permissions to push to the repository `Hagb/decryptBooxUpdateUpx.` Here are steps to resolve this:

1. Verify that the user `RenateUSB` has the correct permissions to the repository. You need at least "write" permissions to push to the repository
2. For PAT, ensure the remote URL includes the correct token. For SSH, the remote should be in the format `git@github.com:Hagb/decryptBooxUpdateUpx.git`
3. The Personal Access Token should have the repo scope checked to push code to the repository.
4. Confirm that the SSH key is associated with the _RenateUSB_ GitHub account and that the key is added to the SSH agent.

If permissions and remote URLs are correct, and you still encounter issues, consider reaching out to the repository owner to ensure your user has the necessary permissions and there are no branch protection rules preventing the push

**Comment by RenateUSB** [comment, marked as answer]
Thanks!
The problem was that I thought that pull requests were done from a separate branch on the main repo.
Apparently you have to make your own repo.
